src/Symfony/Component/Validator/Constraints/DivisibleByValidator.php

```diff
@@ -23,7 +23,26 @@ class DivisibleByValidator extends AbstractComparisonValidator
      */
     protected function compareValues($value1, $value2)
     {
-        return (float) 0 === fmod($value1, $value2);
+        $value1 = \abs($value1);
+        $value2 = \abs($value2);
+        $epsilon = 0.0000001;
+
+        // can't divide by 0
+        if ($value2 < $epsilon) {
```

Comment:
Shouldn't this trigger an exception when the constraint is configured with such an invalid value?

Reply:
Not sure. The previous implementation with fmod just returned NaN (so the validator failed). 

We can change it, I just adapted the previous behavior.